### PR TITLE
[IIIF-928] Add large image feature flag

### PR DESCRIPTION
**Pull request by ksclarke**
* Add Moirai feature flag library
* Add toggles to skip particular types of tests (functional, unit, etc.)
* Add default to clean up containers when tests fail
* Reserve additional ports for spinning up two addition feature flag test containers
* Add correct / dynamically recorded versioning to the OpenAPI spec
* Remove obsolete Codacy configuration (it's now done through profiles)
* Create two additional test containers, one for each feature flag test
* Add status of which features are turned on to the status endpoint
* Create tests to check status endpoint for feature flag configuration

### CODE REVIEW

**Comment by DRickard** [comment]
Small thing, but in GetStatusHandler.java, line 36,
`response.setStatusCode(200);`
why not use `HTTP.OK`?

**Review by DRickard — DISMISSED** [review]

File: pom.xml [hunk 1]
```diff
@@ -659,10 +766,70 @@
...
+          <excludes>
+            <exclude>**/*Test.java</exclude>
+            <exclude>**/*KakaduIT.java</exclude>
+          </excludes>
+          <systemPropertyVariables>
+            <vertx.logger-delegate-factory-class-name>io.vertx.core.logging.SLF4JLogDelegateFactory</vertx.logger-delegate-factory-class-name>
+            <vertx-config-path>${project.basedir}/target/test-classes/test-config.properties</vertx-config-path>
+            <jdk.tls.client.protocols>TLSv1.2</jdk.tls.client.protocols>
+            <jdwp.client.config>${jdwp.client.config}</jdwp.client.config>
+            <jdwp.host.port>${jdwp.host.port}</jdwp.host.port>
```
Comment (markmatney) [inline pom.xml:788]: I'm not able to figure out how to attach a debugger to these containers, btw. Like you said this is going to require a task ticket

**Review by markmatney — CHANGES_REQUESTED** [review]
Do you mind updating the README with info on how to run the tests & use the -Dskip*Ts flags?

File: pom.xml (hunk 1 above)
Comment (ksclarke) [reply pom.xml:788]: Do you think it's best to remove this config info until that time, so as not to give the impression that we already know how to do this?

File: src/test/java/edu/ucla/library/bucketeer/utils/LargeImageFfOffT.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        final Async asyncTask = aContext.async();
+
+        webClient.get(port, Constants.UNSPECIFIED_HOST, "/status").send(statusCheck -> {
+            if (statusCheck.succeeded()) {
+                final HttpResponse<Buffer> response = statusCheck.result();
+                final JsonObject status = response.bodyAsJsonObject();
+                final JsonObject features = status.getJsonObject(Features.FEATURES);
+
+                aContext.assertEquals(status.getString(Constants.STATUS), "ok");
+                aContext.assertEquals(true, features.getBoolean(Features.ENABLED, false));
```
[the remark this answers is not in the document]
Comment (ksclarke) [reply src/test/java/edu/ucla/library/bucketeer/utils/LargeImageFfOffT.java:51]: Oh yes, go ahead and add that to the README in this PR.

File: pom.xml (hunk 1 above)
Comment (markmatney) [reply pom.xml:788]: yeah I think so

File: src/test/java/edu/ucla/library/bucketeer/utils/LargeImageFfOffT.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ksclarke) [reply src/test/java/edu/ucla/library/bucketeer/utils/LargeImageFfOffT.java:51]: Features here mean whether features in general are configured (whether there is a features configuration file found). That's separate from whether an individual feature is turned on or off. Functionally, no features config file found and the feature being turned off are the same thing.

Do that answer the question? I'm not sure if I'm interpreting what you're asking correctly.

[the remark this answers is not in the document]
Comment (markmatney) [reply src/test/java/edu/ucla/library/bucketeer/utils/LargeImageFfOffT.java:51]: yes, thanks

File: pom.xml (hunk 1 above)
Comment (ksclarke) [reply pom.xml:788]: Do you want to do this with your README change, or I can... either way.

**Review by markmatney — APPROVED** [review]
The follow-up ticket for setting up debugging is IIIF-967.

**Review by DRickard — APPROVED** [review]

**Pull request merged**